cirq-google/cirq_google/engine/simulated_local_job.py

```diff
@@ -98,22 +110,40 @@ def batched_results(self) -> Sequence[Sequence[cirq.Result]]:
                 raise e
         raise ValueError('Unsupported simulation type {self._type}')
 
+    def _execute_results(self) -> Sequence[cirq.Result]:
```

Comment:
This method may be called by a separate thread if running asynchronously, which introduces potential thread safety issues for other threads that may be polling the state. Is this a concern?

Reply:
What thread safety issues are you worried about?  Do you think State should be protected by a lock of some sort?Assignment here should be atomic.